src/components/EnumSwitch.ts

```diff
@@ -0,0 +1,141 @@
+            position: 0,
+            width: 0,
+            height: 0,
+            visibility: "visible"
+        };
+        this.enumToggleSlider = this.enumToggleSlider.bind(this);
+        this.createSpan = this.createSpan.bind(this);
+        this.getActiveSpanNodeRef = this.getActiveSpanNodeRef.bind(this);
+        this.WidgetContainerNodeRef = this.WidgetContainerNodeRef.bind(this);
+        this.getButtonNodeRef = this.getButtonNodeRef.bind(this);
+        this.registerEvents = this.registerEvents.bind(this);
+        this.removeEvents = this.removeEvents.bind(this);
+    }
+    render() {
+        if (this.props.status !== "noContext") {
+            return createElement("div", { className: "form-validation" },
+                createElement("div", {
+                    className: classNames("widget-enum-switch", "form-control",
+                        { disabled: this.props.status !== "enabled" }),
+                    ref: this.WidgetContainerNodeRef
+                }, this.createSpan()),
+                createElement(Alert, { message: this.props.alertMessage || "", bootstrapStyle: "danger" })
+            );
+        } else {
+            return createElement("span", { className: "enum-switch noContext" });
+        }
+    }
+
+    private createSpan(): ReactElement<{}>[] {
+        const btnElement: ReactElement<{}>[] = [];
+        if (this.props.enumAttributeValue) {
+            btnElement.push(createElement(EnumButton, {
+                    key: "enumButton",
+                    status: this.props.status,
+                    bootstrapStyle: this.props.bootstrapStyle,
+                    getButtonNode: this.getButtonNodeRef,
+                    position: this.state.position,
+                    visibility: this.state.visibility,
+                    width: this.state.width,
+                    height: this.state.height
+                }));
+            this.registerEvents();
+        }
+        this.props.enumList.forEach((elements, index) => {
+            btnElement.push(createElement(EnumSpan, {
+                key: index,
+                enumAttributeValue: this.props.enumAttributeValue,
+                status: this.props.status,
+                onClickAction: () => this.props.onClickAction(elements.caption),
+                getActiveSpanNode: this.getActiveSpanNodeRef,
+                caption: elements.caption
+            }));
+        });
+
+        return btnElement;
+    }
+
+    private getActiveSpanNodeRef(node: HTMLSpanElement) {
+        this.activeSpanNode = node;
+    }
+
+    private WidgetContainerNodeRef(node: HTMLDivElement) {
+        this.widgetContainerNode = node;
+    }
+
+    private getButtonNodeRef(node: HTMLButtonElement) {
+        this.buttonNode = node;
+    }
+
+    private enumToggleSlider() {
+        const widgetContainer = this.widgetContainerNode;
+        const activeSpan = this.activeSpanNode;
+
+        if (widgetContainer) {
+            if (activeSpan) {
```

Comment:
combine the if statement `if (widgetContainer && activeSpan) {`

Reply:
Active span is a nested if.

Reply:
done